packages/serverpod_client/lib/src/serverpod_client_browser.dart

```diff
@@ -70,28 +52,10 @@ abstract class ServerpodClient extends ServerpodClientShared {
         );
       }
 
-      T result;
-      if (T == getType<void>()) {
-        result = returnVoid() as T;
-      } else {
-        result = parseData<T>(data, T, serializationManager);
-      }
-
-      onSucceededCall?.call(callContext);
-      return result;
-    } catch (e, s) {
-      onFailedCall?.call(callContext, e, s);
-
-      if (e is http.ClientException) {
-        var message = data ?? 'Unknown server response code. ($e)';
-        throw (ServerpodClientException(message, -1));
-      }
-
-      if (logFailedCalls) {
-        print('Failed call: $endpoint.$method');
-        print('$e');
-      }
-      rethrow;
+      return data;
+    } on http.ClientException catch (e) {
+      var message = 'Unknown server response code. ($e)';
+      throw (ServerpodClientException(message, -1));
```

Comment:
This behavior will be subtly different with this refactoring. Before, the ServerpodClientException would be thrown immediately without performing the latter part of the catch clause. Now, it will be handled by the full catch clause (which is now moved to the parent method) _just like all other exceptions_, which seems sensible.

Reply:
Yes and the data var is guaranteed to be null here I believe.